ambry-test-utils/src/main/java/com/github/ambry/quota/ServeEnforcementRecommendation.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+/**
+ * {@link EnforcementRecommendation} implementation with serve recommendation for tests.
+ */
+public class ServeEnforcementRecommendation implements EnforcementRecommendation {
```

Comment:
As I went through different implementations of `EnforcementRecommendation`, I realized that all those implementations are essentially the same. We can just have one class `AmbryEnforcementRecommendation` and then create several static instances, something like
```
public class AmbryEnforcementRecommendation {

  public static final AmbryEnforcementRecommendation REJECT429 = new AmbryEnforcementRecommendation(true, 101, "REJECT", 429, null);
  public static final AmbryEnforcementRecommendation REJECT500 = new AmbryEnforcementRecommendation(true, 101, "REJECT", 500, null);
  public static final AmbryEnforcementRecommendation ALLOW200 = new AmbryEnforcementRecommendation(false, 0, "ALLOW", 200, null);
  public static final AmbryEnforcementRecommendation ALLOW201 = new AmbryEnforcementRecommendation(false, 0, "ALLOW", 201, null);

  private final boolean shouldThrottle;
  private final float usagePercentage;
  private final String quotaEnforcerName;
  private final int recommendedHttpStatus;
  private final RequestCost requestCost;

  public AmbryEnforcementRecommendation(boolean shouldThrottle, float usagePercentage, String quotaEnforcerName,
      int recommendedHttpStatus, RequestCost requestCost) {
    this.shouldThrottle = shouldThrottle;
    this.usagePercentage = usagePercentage;
    this.quotaEnforcerName = quotaEnforcerName;
    this.recommendedHttpStatus = recommendedHttpStatus;
    this.requestCost = requestCost;
  }

  @Override
  public boolean shouldThrottle() {
    return shouldThrottle;
  }

  @Override
  public float getQuotaUsagePercentage() {
    return usagePercentage;
  }

  @Override
  public String getQuotaEnforcerName() {
    return quotaEnforcerName;
  }

  @Override
  public int getRecommendedHttpStatus() {
    return recommendedHttpStatus;
  }

  @Override
  public RequestCost getRequestCost() {
    return requestCost;
  }
}
```